### ISSUE TITLE
GDL segfaults when journaling to a file inside a not existing folder

### ISSUE THREAD

**Issue by brandy125** [gnudatalanguage/gdl#1524]
Downloaded todays version and noticed a segfault when journaling to a file inside a folder that does not exist.

Can anybody confirm this?

```
chris@localhost:~> gdl
  GDL - GNU Data Language, Version 1.0.2 Git
- For basic information type HELP,/INFO
- Default library routine search path used (GDL_PATH/IDL_PATH env. var. not set): /usr/local/bin/../share/gnudatalanguage/lib
- Using WxWidgets as graphics library (windows and widgets).
- Using local drivers in /usr/local/bin/../share/gnudatalanguage/drivers
- Please report bugs, feature or help requests and patches at: https://github.com/gnudatalanguage/gdl

GDL> !gdl
{
    "RELEASE": "1.0.2 Git",
    "BUILD_DATE": "Feb 25 2023",
    "EPOCH": 1677294000,
    "GDL_USE_DSFMT": 1,
    "GDL_USE_WX": 1,
    "GDL_POSIX": 1
}
GDL> journal,'directory_that_does_not_exist/test.jou'
Unhandled GDL exception: File unit is not open.
GDL> .run plot
% Error opening file. File: plot.
% File unit is not open.
terminate called after throwing an instance of 'GDLIOException'
Aborted (core dumped)
chris@localhost:~> 
```

System:

```
Linux localhost.localdomain 6.1.12-1-default #1 SMP PREEMPT_DYNAMIC Wed Feb 15 05:31:41 UTC 2023 (373f017) x86_64 x86_64 x86_64 GNU/Linux
```

```
chris@localhost:~> cat /etc/os-release 
NAME="openSUSE Tumbleweed"
# VERSION="20230223"
ID="opensuse-tumbleweed"
ID_LIKE="opensuse suse"
VERSION_ID="20230223"
PRETTY_NAME="openSUSE Tumbleweed"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:opensuse:tumbleweed:20230223"
BUG_REPORT_URL="https://bugs.opensuse.org"
HOME_URL="https://www.opensuse.org/"
DOCUMENTATION_URL="https://en.opensuse.org/Portal:Tumbleweed"
LOGO="distributor-logo-Tumbleweed"
```

**Comment by alaingdl** [comment]
I cannot reproduce the segfault, but it is clear that we have to cleanup the code in the case the file cannot be opened since GDL print again and again the same message within every subsequent instruction ! 
```
Unhandled GDL exception: File unit is not open.
```

IDL issued only one time the following message
```
IDL> journal,'directory_that_does_not_exist/test.jou'
% JOURNAL: Error opening file. Unit: 100, File: directory_that_does_not_exist/test.jou
  No such file or directory
% Execution halted at: $MAIN$          
```

**Comment by alaingdl** [comment]
Correction ! When just issuing commands, no problem, but if compiling (.r test) then running the code (test) then I have a crash !

**Comment by GillesDuvert** [comment]
solved by #1527
